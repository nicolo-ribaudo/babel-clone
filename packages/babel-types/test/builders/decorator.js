import * as t from "../..";

describe("builders", function() {
  describe("decorator", function() {
    it("legacy", function() {
      expect(t.decorator(t.identifier("foo"))).toMatchSnapshot();
    });

    it("static - no args", function() {
      expect(t.decorator(t.decoratorIdentifier("foo"))).toMatchSnapshot();
    });

    it("static - args", function() {
      expect(
        t.decorator(t.decoratorIdentifier("foo"), [t.identifier("bar")]),
      ).toMatchSnapshot();
    });
  });
});
