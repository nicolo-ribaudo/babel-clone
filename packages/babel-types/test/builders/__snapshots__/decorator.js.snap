// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builders decorator legacy 1`] = `
Object {
  "expression": Object {
    "name": "foo",
    "type": "Identifier",
  },
  "type": "Decorator",
}
`;

exports[`builders decorator static - args 1`] = `
Object {
  "arguments": Array [
    Object {
      "name": "bar",
      "type": "Identifier",
    },
  ],
  "id": Object {
    "name": "foo",
    "type": "DecoratorIdentifier",
  },
  "type": "Decorator",
}
`;

exports[`builders decorator static - no args 1`] = `
Object {
  "arguments": undefined,
  "id": Object {
    "name": "foo",
    "type": "DecoratorIdentifier",
  },
  "type": "Decorator",
}
`;
